src/cpp/src/llm_pipeline.cpp

```diff
@@ -0,0 +1,438 @@
+    auto config_arg = get_config_from_map(config_map);
+    GenerationConfig config = (config_arg.has_value()) ? *config_arg : get_generation_config();
+    config.update_generation_config(config_map);
+
+    return m_pimpl->generate(inputs, config, get_streamer_from_map(config_map));
+}
+
+std::pair<std::string, Any> streamer(StreamerVariant func) {
+    if (auto streamer_obj = std::get_if<std::shared_ptr<StreamerBase>>(&func)) {
+        return {STREAMER_ARG_NAME, Any::make<std::shared_ptr<StreamerBase>>(*streamer_obj)};
+    } else  {
+        auto callback = std::get<std::function<bool(std::string)>>(func);
+        return {STREAMER_ARG_NAME, Any::make<std::function<bool(std::string)>>(callback)};
+    } 
+}
+
+std::pair<std::string, Any> generation_config(const GenerationConfig& config) {
+    return {CONFIG_ARG_NAME, Any::make<GenerationConfig>(config)};
+}
+
+}  // namespace genai
+}  // namespace ov
+
+using namespace std;
+
+ov::genai::LLMPipeline::LLMPipeline(
+    const ov::InferRequest& request, 
+    const ov::genai::Tokenizer& tokenizer, 
+    OptionalGenerationConfig generation_config
+) {
+    m_pimpl = std::make_unique<LLMPipelineImpl>(request, tokenizer, generation_config);
+}
+
+
+ov::genai::LLMPipeline::LLMPipeline(
+    const std::string& model_path,
+    const ov::genai::Tokenizer& tokenizer,
+    const std::string& device,
+    const ov::AnyMap& plugin_config
+) {
+    m_pimpl = make_unique<LLMPipelineImpl>(std::filesystem::path(model_path), tokenizer, device, plugin_config);
+}
+
+ov::genai::LLMPipeline::LLMPipelineImpl::LLMPipelineImpl(
+    const std::filesystem::path& model_path,
+    const ov::genai::Tokenizer& tokenizer,
+    const std::string& device,
+    const ov::AnyMap& plugin_config
+): m_tokenizer(tokenizer) {
+    ov::Core core;
+    
+    std::filesystem::path full_path = model_path;
+    if (full_path.extension() != ".xml")
+        full_path = model_path / "openvino_model.xml";
+    m_model_runner = core.compile_model(full_path, device, plugin_config).create_infer_request();
+}
+
+ov::genai::LLMPipeline::LLMPipeline(
+    const std::string& path, 
+    const std::string& device, 
+    const ov::AnyMap& config
+) {
+    m_pimpl = make_unique<LLMPipelineImpl>(std::filesystem::path(path), device, config);
+}
+
+ov::genai::LLMPipeline::LLMPipelineImpl::LLMPipelineImpl(
+    const std::filesystem::path& path, 
+    const std::string& device, 
+    const ov::AnyMap& config
+): 
+    m_model_runner{ov::Core{}.compile_model(path / "openvino_model.xml", device, config).create_infer_request()}, 
+    m_tokenizer(path.string()),
+    m_generation_config{from_config_json_if_exists(path)},
+    m_chat_template{chat_template_from_tokenizer_json_if_exists(path)}
+{
+    // If eos_token_id was not provided, take value
+    if (m_generation_config.eos_token_id == -1)
+        m_generation_config.eos_token_id = m_tokenizer.get_eos_token_id();
```

Comment:
@pavel-esir 
should we do it in other ctors as well?

Reply:
done